src/main/java/InfrastructureManager/Modules/NetworkStructure/Shared/SharedDistance.java

```diff
@@ -0,0 +1,26 @@
+package InfrastructureManager.Modules.NetworkStructure.Shared;
+
+
+import InfrastructureManager.ModuleManagement.ImmutablePlatformModule;
+import InfrastructureManager.Modules.NetworkStructure.NetworkModuleObject;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+
+public class SharedDistance extends NetworkModuleObject {
+	private final BlockingQueue distanceQueue;
+
+	public SharedDistance(ImmutablePlatformModule ownerModule) {
+		super(ownerModule);
+		this.distanceQueue = new ArrayBlockingQueue<>(25);
```

Comment:
I believe ArrayBlockingQueue is buffer bound, which means it has a limited size (25 here), why don't you use an unlimited queue?
Also why don't you define the type? (it is double).